SHMEM/isx.c

```diff
@@ -722,106 +738,99 @@ static void print_run_info(FILE * fp)
   }
 }
 
-/* 
+/*
  * Aggregates the per PE timing information
- */ 
+ */
 static double * gather_rank_times(_timer_t * const timer)
 {
   if(timer->seconds_iter > 0) {
 
     assert(timer->seconds_iter == timer->num_iters);
 
-    const unsigned int num_records = NUM_PES * timer->seconds_iter;
-#ifdef OPENSHMEM_COMPLIANT
-    double * my_times = shmem_malloc(timer->seconds_iter * sizeof(double));
-#else
-    double * my_times = shmalloc(timer->seconds_iter * sizeof(double));
-#endif
-    memcpy(my_times, timer->seconds, timer->seconds_iter * sizeof(double));
+    shmem_barrier_all();
 
 #ifdef OPENSHMEM_COMPLIANT
-    double * all_times = shmem_malloc( num_records * sizeof(double));
+    double * my_average_time = shmem_malloc(sizeof(double));
 #else
-    double * all_times = shmalloc( num_records * sizeof(double));
+    double * my_average_time = shmalloc(sizeof(double));
 #endif
+    double temp = 0.0;
+    for(unsigned int i = 0; i < timer->seconds_iter; i++) {
+      temp += timer->seconds[i];
+    }
 
-    shmem_barrier_all();
+    *my_average_time = temp/(timer->seconds_iter);
+
+    double * pe_average_times = shmem_malloc( NUM_PES * sizeof(double));
 
-    shmem_fcollect64(all_times, my_times, timer->seconds_iter, 0, 0, NUM_PES, pSync);
+    shmem_barrier_all();
+    shmem_fcollect64(pe_average_times, my_average_time, 1, 0, 0, NUM_PES, pSync);
     shmem_barrier_all();
 
-#ifdef OPENSHMEM_COMPLIANT
-    shmem_free(my_times);
-#else
-    shfree(my_times);
-#endif
+    shmem_free(my_average_time);
 
-    return all_times;
+    return pe_average_times;
   }
+
   else{
     return NULL;
   }
 }
 
 /*
- * Aggregates the per PE timing 'count' information 
+ * Aggregates the per PE timing 'count' information
  */
 static unsigned int * gather_rank_counts(_timer_t * const timer)
 {
-  if(timer->count_iter > 0){
-    const unsigned int num_records = NUM_PES * timer->num_iters;
+  if(timer->count_iter > 0) {
 
-#ifdef OPENSHMEM_COMPLIANT
-    unsigned int * my_counts = shmem_malloc(timer->num_iters * sizeof(unsigned int));
-#else
-    unsigned int * my_counts = shmalloc(timer->num_iters * sizeof(unsigned int));
-#endif
-    memcpy(my_counts, timer->count, timer->num_iters*sizeof(unsigned int));
+    shmem_barrier_all();
 
 #ifdef OPENSHMEM_COMPLIANT
-    unsigned int * all_counts = shmem_malloc( num_records * sizeof(unsigned int) );
+    unsigned int * my_average_count = shmem_malloc(sizeof(unsigned int));
 #else
-    unsigned int * all_counts = shmalloc( num_records * sizeof(unsigned int) );
+    unsigned int * my_average_count = shmalloc(sizeof(unsigned int));
 #endif
+    unsigned int temp = 0;
+    for(unsigned int i = 0; i < timer->count_iter; i++) {
+      temp += timer->count[i];
+    }
 
-    shmem_barrier_all();
+    *my_average_count = temp/(timer->count_iter);
 
-    shmem_collect32(all_counts, my_counts, timer->num_iters, 0, 0, NUM_PES, pSync);
+    unsigned int * pe_average_counts = shmem_malloc( NUM_PES * sizeof(unsigned int));
```

Comment:
Minor spacing suggestion:

```suggestion
    unsigned int * pe_average_counts = shmem_malloc(NUM_PES * sizeof(unsigned int));
```